specs/capri/object_lifecycle_management.md

```diff
@@ -0,0 +1,568 @@
+    <Transition>
+      <Days>365</Days>
+      <StorageClass>GLACIER</StorageClass>
+    </Transition>
+    <Expiration>
+      <Days>3650</Days>
+      </Expiration>
+  </Rule>
+</LifecycleConfiguration>
+```
+### DELETE Bucket Lifecycle
+The DELETE Bucket Lifecycle operations removes the lifecycle configurations set on a bucket. To use this operation, you must have permission to perform the PUT Bucket lifecycle action.
+The DELETE Bucket Lifecycle operation does not use request parameters and elements. More than one lifecycle rule deletion is allowed in OpenSDS.
+
+Request Syntax:
+```
+DELETE bucketname/?lifecycle HTTP/1.1
+Host: x.x.x.x
+Date: date
+Authorization: authorization string
+```
+Curl Syntax:
+```
+DELETE http://{host_ip}/v1/s3/<bucket_name>/?lifecycle&ruleID=<ruleID1>&ruleID=<ruleID2>...&ruleID=<ruleIDn>
+```
+Response Sample :
+```
+HTTP/1.1 200 OK
+x-amz-id-2: Uuag1LuByRx9e6j5OnimrSAMPLEtRPfTaOAa==  
+x-amz-request-id: 656c76696e672SAMPLE5657374  
+Date: Wed, 14 Dec 2011 05:37:16 GMT
+Connection: keep-alive  
+Server: x.x.x.x   
+```
+### Recover Object from Glacier Storage(Get the recover progress)
+Objects in the GLACIER storage class are archived. To access an archived object, you must first initiate a restore request. This restores a copy of the archived object. The time it takes restore jobs to finish depends on which data access tier you specify, Expedited, Standard, or Bulk.
+Speed Option
+When restoring an archived object (or using a select request), you can specify one of the following options in the Tier element of the request body:
+- Expedited – Lets you quickly access your data when occasional urgent requests for a subset of archives are required. For all but the largest archived object (250 MB+), data accessed using Expedited retrievals are typically made available within 1–5 minutes.
+- Standard – Lets you access any of your archived objects within several hours. Standard retrievals typically finish within 3–5 hours. This is the default tier.
+- Bulk – The lowest-cost data access option in Glacier. It lets you retrieve large amounts, even petabytes, of data inexpensively in a day. Bulk access typically completes within 5–12 hours.
+Important: OpenSDS does not guarantee those recover time for each option, because some cloud storage service does not support those options, some other only support a part of those options.
+- Azure blob does not support those options.
+- Huawei OBS, only support Expedited and Standard. So when you specify Bulk, then Standard will be used.
+Lifetime of the restored (active) copy
+In a restore request, you specify the number of days that you want the restored copy to exist.
+But it only used for those backend which support restore period. For:
+Amazon S3 and Huawei OBS, after the specified period, the restored copy will be deleted, while the object remains archived.
+- Azure, it is not used, after recovered, the object will be stored in the new storage class, not in Glacier any more.
```

Comment:
Rather than saying GLACIER, should we say t999 to make it consistent in context of OpenSDS

Reply:
Agree, we need to keep consistent in the whole design spec.